src/test/java/org/opensearch/security/SecurityRolesTests.java

```diff
@@ -93,6 +93,24 @@ public void testSecurityRoles() throws Exception {
 		Assert.assertEquals(HttpStatus.SC_OK, resc.getStatusCode());
 	}
 
+	@Test
+	public void testRecursiveActionGroup() throws Exception {
+		// ConfigUpdateResponse is expected to fail and throw AssertionError
+		Assert.assertThrows(AssertionError.class, () -> {
+		setup(Settings.EMPTY, new DynamicSecurityConfig()
+				.setSecurityActionGroups("action_groups_recursive.yml")
+				.setSecurityRoles("roles_recursive.yml")
+				.setSecurityRolesMapping("roles_mapping_recursive.yml")
+				.setSecurityInternalUsers("internal_users_recursive.yml"), Settings.EMPTY, true);
+		});
+
+		RestHelper rh = nonSslRestHelper();
+		rh.sendAdminCertificate = false;
+
+		HttpResponse resc = rh.executeGetRequest("_cat/nodes?pretty", encodeBasicHeader("recursive_internal_user", "nagilum"));
```

Comment:
Not a proper test, it succeeds even without code changes in `ConfigModelV7.java`.